### PR TITLE
Add full vitepress documentation page

### PR DESCRIPTION
**Pull request by jkniest**

### CODE REVIEW

File: docs/.vitepress/config.mts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          { text: 'Create a product', link: '/examples/product' },
+        ],
+      },
+      //{
+      //  text: 'Contributing',
+      //  items: [
+      //    { text: 'Internals', link: '/contributing/internals' },
+      //    { text: 'Contribution guide', link: '/contributing/guide' },
+      //  ]
+      //}
```
Comment (rvalley98) [inline docs/.vitepress/config.mts:69]: What is this still needed for

File: Makefile [hunk 2]
```diff
@@ -44,7 +44,7 @@ shell:
 ## Install all dependencies
 dependencies:
 	make docker COMMAND="composer install --no-interaction --optimize-autoloader"
-	make docker COMMAND="npm ci"
+	make docker COMMAND="npm install"
```
Comment (rvalley98) [inline Makefile:47]: Why was the clean install removed? wouldn't it be safer to perform the clean install?

**Review by rvalley98 — APPROVED** [review]

File: docs/.vitepress/config.mts (hunk 1 above)
Comment (jkniest) [reply docs/.vitepress/config.mts:69]: I will add the pages afterwards to the docs, that why I left them in :D Just didn't get to it before the V3 release (and its not too iumportant for a major release)

File: Makefile (hunk 2 above)
Comment (jkniest) [reply Makefile:47]: Its only to run prettier tbh so it doesn't really change much. Reason being that I swapped the vitepress project to bun so no "package-lock.json" file exists anymore

**Pull request merged**